user/views.py

```diff
@@ -206,3 +208,28 @@ def update_password(user, validated_data):
                 data={"message": "Old password is incorrect.", "code": 400},
                 status=status.HTTP_400_BAD_REQUEST,
             )
+
+
+class UserRoadmapDetailAPIView(RetrieveAPIView):
+    queryset = UserRoadmap.objects.all()
+    serializer_class = UserRoadmapDetailSerializer
+    authentication_classes = [CustomJWTAuthentication,]
+    lookup_field = "user"
+    lookup_url_kwarg = "user_pk"
+
+    def get_queryset(self):
+        queryset = self.queryset
+        user = self.request.user
+        roadmap_pk = self.kwargs.get("roadmap_pk", None)
+        if roadmap_pk is not None:
+            queryset = queryset.filter(user_id=user.id, roadmap_id=roadmap_pk)
+
+        return queryset
+
+    def get_object(self):
+        try:
+            obj = super().get_object()
+        except Http404 as e:
+            raise NotFound({"message": str(e), "code": 404})
+        else:
+            return obj
```

Comment:
let's use handle_exception method (https://www.cdrf.co/3.1/rest_framework.generics/RetrieveAPIView.html#handle_exception)
check i`f isinstance(exc, Http404) `then raise `NotFound({"message": "User Roadmap not found", "code": 404})`